pkg/datapath/loader/loader.go

```diff
@@ -325,7 +325,7 @@ func (l *Loader) reloadHostDatapath(ctx context.Context, ep datapath.Endpoint, o
 	// Replace program on cilium_net.
 	if _, err := netlink.LinkByName(defaults.SecondHostDevice); err != nil {
 		log.WithError(err).WithField("device", defaults.SecondHostDevice).Error("Link does not exist")
-		return err
+		return fmt.Errorf("device '%s' not found: %w", defaults.SecondHostDevice, err)
```

Comment:
Nit: this is really unlikely to happen, as we create this device a little bit before this, but I guess that's where it went wrong while coming up with this patch. :)